### DISCUSSION TITLE
Having an alternating style for odd or even pages

### DISCUSSION THREAD

**Discussion by PedroCavalheiro** [marp-team/marp#456] [Q&A]
Hello,

In print, it's usual for the page marker to alternate between the right and left side of the page. Is there a way to apply this in Marp?

The expected section:nth-child solution doesn't work since each section is a standalone element.

Thank you for your attention.

**Comment by yhatt** [comment]
Marp is designed for PowerPoint style presentation slides, and is not supported the book style pagination.

If necessary, you can write a Marp (Marpit) plugin to add `odd` or `even` class to the `<section>` element, and use it in [Marp CLI](https://github.com/marp-team/marp-cli).

```javascript
// engine.mjs
const marpOddEvenSlidePlugin = (md) => {
  md.core.ruler.after('marpit_slide', 'marp_odd_even_slide', (state) => {
    if (state.inlineMode) return

    for (const token of state.tokens) {
      if (token.type === 'marpit_slide_open') {
        const pageNumber = token.meta.marpitSlide + 1
        token.attrJoin('class', pageNumber % 2 === 0 ? 'even' : 'odd')
      }
    }
  })
}

export default ({ marp }) => marp.use(marpOddEvenSlidePlugin)
```

```
marp --engine engine.mjs markdown.md
```

**Comment by PedroCavalheiro** [comment]
Since this is the first Google result and I have come across the problem again (and was surprised to realize I had already asked it a year ago), here's the CSS selector to target odd pages, which can be adapted for even pages:
```css
section[data-marpit-pagination$="1"],
section[data-marpit-pagination$="3"],
section[data-marpit-pagination$="5"],
section[data-marpit-pagination$="7"],
section[data-marpit-pagination$="9"]{
  /* style goes here */
}
```
